tests/acceptance/TestHelpers/TusClient.php

```diff
@@ -0,0 +1,138 @@
+<?php declare(strict_types=1);
+/**
+ * ownCloud
+ *
+ * @author Nabin Magar <nabin@jankaritech.com>
+ * @copyright Copyright (c) 2025 Nabin Magar nabin@jankaritech.com
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License,
+ * as published by the Free Software Foundation;
+ * either version 3 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace TestHelpers;
+
+use Carbon\Carbon;
+use GuzzleHttp\Exception\ClientException;
+use GuzzleHttp\Exception\GuzzleException;
+use TusPhp\Exception\ConnectionException;
+use TusPhp\Exception\FileException;
+use Psr\Http\Message\ResponseInterface;
+use Symfony\Component\HttpFoundation\Response as HttpResponse;
+use TusPhp\Exception\TusException;
+use TusPhp\Tus\Client;
+
+/**
+ * A TUS client based on TusPhp\Tus\Client
+ */
+
+class TusClient extends Client {
+
+	/**
+	 * creates a resource with a post request and returns response.
+	 *
+	 * @param string $key
+	 * @param int $bytes
+	 *
+	 * @return ResponseInterface
+	 * @throws GuzzleException
+	 */
+	public function createWithUploadRR(string $key, int $bytes = -1): ResponseInterface {
```

Comment:
this name sounds so odd createWithUploadRR
```suggestion
	public function createWithUploadRR(string $key, int $bytes = -1): ResponseInterface {
```